### PR TITLE
Enable AnalysisLevel latest-All

### LINKED ISSUE #950: Enable AnalysisLevel=latest-All and fix resulting issues
# Feature Request

**Is your feature request related to a problem?**

As originally suggested by https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/900#issuecomment-1387222338, enable `AnalysisLevel=latest-All` for all of the assemblies to find and then fix common problematic/buggy code patterns.

**Describe the solution you'd like:**

Set `<AnalysisLevel>latest-All</AnalysisLevel>` in `build/Common.props` and fix all of the resulting issues.

**Describe alternatives you've considered.**

None.

## Additional Context

Initially implemented as #908, but requested to be carved into multiple pull requests by https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/908#issuecomment-1396955313.

PRs to port the changes from #908 project-by-project to follow.

### PR DESCRIPTION
**Pull request by martincostello**
## Changes

- Enable `AnalysisLevel` at `latest-All`.
- Add relevant suppressions for test projects.

Resolves #950.

Depends on:

- #951
- #952
- #953
- #954
- #955
- #956
- #957
- #958
- #959
- #960
- #961
- #962
- #963
- #964
- #965
- #966
- #967
- #968
- #1043
- #1061

### CODE REVIEW

**Review by Kielek — APPROVED** [review]

File: build/OpenTelemetryContrib.test.ruleset [hunk 1]
```diff
@@ -4,6 +4,36 @@
     <Name Resource="OpenTelemetrySDKRules_Name" />
     <Description Resource="OpenTelemetrySDKRules_Description" />
   </Localization>
+  <Rules AnalyzerId="Microsoft.Analyzers.ManagedCodeAnalysis" RuleNamespace="Microsoft.Rules.Managed">
+    <Rule Id="CA1001" Action="None" />
```
Comment (Kielek) [inline build/OpenTelemetryContrib.test.ruleset:8]: Question for potential follow ups.
Do you think that on this list there is any other important warning?

Comment (martincostello) [reply build/OpenTelemetryContrib.test.ruleset:8]: I don't think there was anything particularly _important_, if I recall correctly.

**Review by utpilla — APPROVED** [review]

**Comment by utpilla** [comment]
@martincostello Thanks a lot for working on this so patiently. #875 was merged before this which made some updates to two of the test files in AWSXRay project. This resulted in some merge conflicts. Could you please resolve them and we can merge this PR?

**Comment by martincostello** [comment]
It would be good if this can be merged (once green) before any other PRs, as other PRs are unintentionally introducing new analysis warnings that I then need to fix here.

**Pull request merged**